usecases/data_model_usecase.go

```diff
@@ -162,42 +183,171 @@ func (usecase *DataModelUseCase) CreateDataModelField(ctx context.Context, field
 	exec := usecase.executorFactory.NewExecutor()
+	// permission and input validation
 	field, err := usecase.dataModelRepository.GetDataModelField(ctx, exec, fieldID)
 	if err != nil {
 		return err
 	}
-
-	if table, err := usecase.dataModelRepository.GetDataModelTable(ctx, exec, field.TableId); err != nil {
+	table, err := usecase.dataModelRepository.GetDataModelTable(ctx, exec, field.TableId)
+	if err != nil {
 		return err
 	} else if err := usecase.enforceSecurity.WriteDataModel(table.OrganizationID); err != nil {
 		return err
 	}
+	dataModel, err := usecase.GetDataModel(ctx, table.OrganizationID)
+	if err != nil {
+		return err
+	}
+
+	makeUnique, makeNotUnique, err := validateFieldUpdateRules(dataModel, field, table, input)
+	if err != nil {
+		return err
+	}
+
+	// update the field (data_model_field row)
+	if err := usecase.dataModelRepository.UpdateDataModelField(ctx, exec, fieldID, input); err != nil {
+		return err
+	}
+
+	// asynchronously create the unique index if required
+	if makeUnique {
+		return usecase.clientDbIndexEditor.CreateUniqueIndexAsync(ctx, models.UnicityIndex{
+			TableName: models.TableName(table.Name),
+			Fields:    []models.FieldName{models.FieldName(field.Name)},
+		})
+	}
+
+	// delete the unique index if required
+	if makeNotUnique {
+		return usecase.clientDbIndexEditor.DeleteUniqueIndex(ctx, models.UnicityIndex{
+			TableName: models.TableName(table.Name),
+			Fields:    []models.FieldName{models.FieldName(field.Name)},
+		})
+	}
+
+	return nil
+}
+
+func validateFieldUpdateRules(
+	dataModel models.DataModel,
+	field models.FieldMetadata,
+	table models.TableMetadata,
+	input models.UpdateFieldInput,
+) (makeUnique, makeNotUnique bool, err error) {
+	makeEnum := input.IsEnum != nil && *input.IsEnum && !field.IsEnum
+	makeNotEnum := input.IsEnum != nil && !*input.IsEnum && field.IsEnum
+	if makeEnum && !slices.Contains(enumTypes, field.DataType) {
+		return false, false, errors.Wrap(
+			models.BadParameterError,
+			"enum fields can only be of type string or numeric")
+	}
+
+	currentField := dataModel.Tables[models.TableName(table.Name)].Fields[models.FieldName(field.Name)]
+
+	makeUnique = input.IsUnique != nil &&
+		*input.IsUnique &&
+		currentField.UnicityConstraint == models.NoUnicityConstraint
+	if makeUnique && !slices.Contains(uniqTypes, field.DataType) {
+		return false, false, errors.Wrap(
+			models.BadParameterError,
+			"unique fields can only be of type string, int or float")
+	}
+
+	linksToField := findLinksToField(dataModel, table.Name, field.Name)
+	makeNotUnique = input.IsUnique != nil &&
+		!*input.IsUnique &&
+		currentField.UnicityConstraint != models.NoUnicityConstraint
+	if makeNotUnique && len(linksToField) > 0 {
+		return false, false, errors.Wrap(
+			models.BadParameterError,
+			"cannot remove unicity constraint on a field that is linked to another table")
+	}
+
+	if makeUnique && ((field.IsEnum && !makeNotEnum) || makeEnum) {
```

Comment:
fundamentally nothing bad would happen if a field were at the same time enum and unique, but it doesn't make sense so I want to prevent it in the front, and I'm adding the same constraint here to avoid the front being in a blocked situation (both fields toggled and disabled)